azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/TemplateToDeploymentTemplateLiveTest.java

```diff
@@ -0,0 +1,196 @@
+        deploymentName = "jc" + now;
+
+        String rsakey = new String("ssh-rsa AAAAB3NzaC1yc2EAAAABJQAAAQEAmfk/QSF0pvnrpdz+Ah2KulGruKU+8FFBdlw938MpOysRdmp7uwpH6Z7+5VNGNdxFIAyc/W3UaZXF9hTsU8+78TlwkZpsr2mzU+ycu37XLAQ8Uv7hjsAN0DkKKPrZ9lgUUfZVKV/8E/JIAs03gIbL6zO3y7eYJQ5fNeZb+nji7tQT+YLpGq/FDegvraPKVMQbCSCZhsHyWhdPLyFlu9/30npZ0ahYOPI/KyZxFDtM/pHp88+ZAk9Icq5owaLRWcJQqrBGWqjbZnHtjdDqvHZ+C0wPhdJZPyfkHOrSYTwSQBXfX4JLRRCz3J1jf62MbQWT1o6Y4JEs1ZP1Skxu6zR96Q== mocktest");
+
+        AzureComputeArmTemplateOptions options = new AzureComputeArmTemplateOptions();
+        options.authorizePublicKey(rsakey);
+        DeploymentTemplateBuilder templateBuilder  = getDeploymentTemplateBuilderWithOptions(options);
+
+        DeploymentBody deploymentTemplateBody =  templateBuilder.getDeploymentTemplate();
+
+        DeploymentProperties properties = DeploymentProperties.create(deploymentTemplateBody);
+        Gson gson = new GsonBuilder().disableHtmlEscaping().create();
+        org.jclouds.json.Json json = new GsonWrapper(gson);
+
+        String deploymentTemplate = json.toJson(properties);
+        deploymentTemplate = UrlEscapers.urlFormParameterEscaper().escape(deploymentTemplate);
+
+        Deployment deployment = api().validate(deploymentName, deploymentTemplate);
+        assertNotNull(deployment);
+    }
+
+    @Test(groups = "live")
+    public void testCreateDeploymentTemplateLinuxNode() {
+        Long now = System.currentTimeMillis();
+        deploymentName = "jc" + now;
+
+        String rsakey = new String("ssh-rsa AAAAB3NzaC1yc2EAAAABJQAAAQEAmfk/QSF0pvnrpdz+Ah2KulGruKU+8FFBdlw938MpOysRdmp7uwpH6Z7+5VNGNdxFIAyc/W3UaZXF9hTsU8+78TlwkZpsr2mzU+ycu37XLAQ8Uv7hjsAN0DkKKPrZ9lgUUfZVKV/8E/JIAs03gIbL6zO3y7eYJQ5fNeZb+nji7tQT+YLpGq/FDegvraPKVMQbCSCZhsHyWhdPLyFlu9/30npZ0ahYOPI/KyZxFDtM/pHp88+ZAk9Icq5owaLRWcJQqrBGWqjbZnHtjdDqvHZ+C0wPhdJZPyfkHOrSYTwSQBXfX4JLRRCz3J1jf62MbQWT1o6Y4JEs1ZP1Skxu6zR96Q== mocktest");
+
+        AzureComputeArmTemplateOptions options = new AzureComputeArmTemplateOptions();
+        options.authorizePublicKey(rsakey);
+        DeploymentTemplateBuilder templateBuilder  = getDeploymentTemplateBuilderWithOptions(options);
+
+        DeploymentBody deploymentTemplateBody =  templateBuilder.getDeploymentTemplate();
+        DeploymentProperties properties = DeploymentProperties.create(deploymentTemplateBody);
+
+        Gson gson = new GsonBuilder().disableHtmlEscaping().create();
+        org.jclouds.json.Json json = new GsonWrapper(gson);
+
+        String deploymentTemplate = json.toJson(properties);
+        deploymentTemplate = UrlEscapers.urlFormParameterEscaper().escape(deploymentTemplate);
+
+        //creates an actual VM using deployment template
+        Deployment deployment = api().create(deploymentName, deploymentTemplate);
+
+        Deployment.ProvisioningState state = Deployment.ProvisioningState.fromString(deployment.properties().provisioningState());
+        int testTime = 0;
+        while (testTime < maxTestDuration) {
```

Comment:
Same comment about polling and reusable predicates bound to the Guice injector. This predicates are used in tests and potentially by the compute service, so please, make them injectable so they can be shared, and the polling parameters configured.